src/main/java/com/faforever/moderatorclient/ui/TutorialAddController.java

```diff
@@ -0,0 +1,142 @@
+import javafx.fxml.FXML;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Pane;
+import javafx.stage.Stage;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.text.MessageFormat;
+
+@Component
+@Scope("prototype")
+@Slf4j
+public class TutorialAddController implements Controller<Pane> {
+    private final TutorialService tutorialService;
+    public TextField descriptionField;
+    public TextField titleField;
+    public TextField imageField;
+    public TextField categoryField;
+    public TextField ordinalField;
+    public TextField mapField;
+    public GridPane root;
+    public Label errorLabel;
+    public CheckBox launchableBox;
+    public TextField technicalNameField;
+
+    private Runnable onSaveRunnable;
+
+    @Inject
+    public TutorialAddController(TutorialService tutorialService) {
+        this.tutorialService = tutorialService;
+    }
+
+    @Override
+    public Pane getRoot() {
+        return root;
+    }
+
+    @FXML
+    public void initialize() {
+        errorLabel.managedProperty().bind(errorLabel.visibleProperty());
+        errorLabel.setVisible(false);
+    }
+
+    public void onSave() {
+        if (!validate()) {
+            return;
+        }
+        TutorialFx tutorialFx = new TutorialFx();
+        tutorialFx.setTitleKey(titleField.getText());
+        tutorialFx.setDescriptionKey(descriptionField.getText());
+        tutorialFx.setLaunchable(launchableBox.isSelected());
+        String mapId = mapField.getText();
+        if(!mapId.isEmpty()){
+            MapVersionFX mapVersionFX = new MapVersionFX();
+            mapVersionFX.setId(mapId);
+            tutorialFx.setMapVersion(mapVersionFX);
+        }
+        int categoryId = Integer.parseInt(categoryField.getText());
+        TutorialCategoryFX tutorialCategoryFX = new TutorialCategoryFX();
+        tutorialCategoryFX.setId(categoryId);
+        tutorialFx.setCategory(tutorialCategoryFX);
+        tutorialFx.setImage(imageField.getText());
+        tutorialFx.setOrdinal(Integer.parseInt(ordinalField.getText()));
+        tutorialFx.setTechnicalName(technicalNameField.getText());
+        try {
+            if (tutorialService.create(tutorialFx) == null) {
+                error("Not saved unknown error");
+                return;
+            }
+        } catch (Exception e) {
+            error(MessageFormat.format("Unable to save Tutorial error is:`{0}`", e.getMessage()));
+            log.warn("Tutorial not saved", e);
+            return;
+        }
+
+        close();
+        if(onSaveRunnable!=null){
+            onSaveRunnable.run();
+        }
+    }
+
+    private boolean validate() {
+        if(titleField.getText().isEmpty()){
```

Comment:
Please catch all errors in a List and concat all of them in a stream to show all of them at once. There are examples in other Controllers.